tests/test_smoke.py

```diff
@@ -1835,20 +1850,27 @@ def test_bucket_bulk_deletion(self, store_type):
                 f'touch {tmpdir}/folder{{000..255}}/test.txt', shell=True)
 
             timestamp = str(time.time()).replace('.', '')
-            store_obj = storage_lib.Storage(name=f'sky-test-{timestamp}',
-                                            source=tmpdir)
+            if store_type == storage_lib.StoreType.R2:
+                store_obj = storage_lib.Storage(name=f'r2-sky-test-{timestamp}',
+                                                source=tmpdir)
+            else:
+                store_obj = storage_lib.Storage(name=f'sky-test-{timestamp}',
+                                                source=tmpdir)
             store_obj.add_store(store_type)
 
         subprocess.check_output(['sky', 'storage', 'delete', store_obj.name])
 
         output = subprocess.check_output(['sky', 'storage', 'ls'])
         assert store_obj.name not in output.decode('utf-8')
-
+    
     @pytest.mark.parametrize(
         'tmp_public_storage_obj, store_type',
         [('s3://tcga-2-open', storage_lib.StoreType.S3),
          ('s3://digitalcorpora', storage_lib.StoreType.S3),
-         ('gs://gcp-public-data-sentinel-2', storage_lib.StoreType.GCS)],
+         ('gs://gcp-public-data-sentinel-2', storage_lib.StoreType.GCS),
+         ('r2://test-bucket', storage_lib.StoreType.R2)],
+         # To test for r2, need to provide existing bucket name
+         # that corresponds to r2 credential and accountid you provide
```

Comment:
That `r2://test-bucket` is just my bucket. Now that you mention, actually that test is wrong as `r2://test-bucket` is my private bucket. But, yeah there is a notion of public bucket in R2. I'll update the test with that.

Reply:
["Public buckets only provide GET/HEAD on a known object path."
](https://developers.cloudflare.com/r2/data-access/s3-api/presigned-urls/#differences-between-presigned-urls-and-public-buckets)
It turns out that the notion of public bucket is a bit different for R2 compared to S3 and GCS. Public R2 bucket only allows access to a specific object stored in the bucket rather than allowing access to the bucket's file system. For example, it's possible to run `aws s3 ls s3://digitalcorpora` where `s3://digitalcorpora` is a public s3 bucket. But it's impossible to run `aws s3 ls` on R2 public bucket url. For R2, allowing public access just means allowing access to a specific object for public rather than the entire bucket's file system.

I'll remove R2 from this test for now.